assets/js/src/core/modules/data-object/actions/paste/use-paste.tsx

```diff
@@ -0,0 +1,125 @@
+  pasteRecursiveUpdatingReferencesTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+  pasteOnlyContentsTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+  isPasteMenuHidden: (node: Element | TreeNodeProps) => boolean
+}
+
+export const usePaste = ({ storedNode, nodeTask }: UsePasteHookParams): UsePasteHookReturn => {
+  const { t } = useTranslation()
+  const { isTreeActionAllowed } = useTreePermission()
+  const dispatch = useAppDispatch()
+  const { paste } = useCopyPaste('data-object')
+  const { treeId } = useTreeId(true)
+
+  const pasteAsChildRecursiveTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-as-child-recursive'),
+      key: 'pasteAsChildRecursive',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOpenHidden(node),
+      onClick: async () => {
+        dispatch(setNodeFetching({ treeId, nodeId: String(node.id), isFetching: true }))
+        await paste(parseInt(node.id), { recursive: true, updateReferences: false }, storedNode)
+      }
+    }
+  }
+
+  const pasteRecursiveUpdatingReferencesTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-recursive-updating-references'),
+      key: 'pasteRecursiveUpdatingReferences',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOpenHidden(node),
+      onClick: async () => {
+        dispatch(setNodeFetching({ treeId, nodeId: String(node.id), isFetching: true }))
+        await paste(parseInt(node.id))
+      }
+    }
+  }
+
+  const pasteAsChildTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-as-child'),
+      key: 'pasteAsChild',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOpenHidden(node),
+      onClick: async () => {
+        dispatch(setNodeFetching({ treeId, nodeId: String(node.id), isFetching: true }))
+        await paste(parseInt(node.id), { recursive: false, updateReferences: false })
+      }
+    }
+  }
+
+  const pasteOnlyContentsTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-only-contents'),
+      key: 'pasteOnlyContents',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOnlyContentsHidden(node),
+      onClick: async () => {
+        console.log('not implemented!')
+      }
+    }
+  }
+
+  const isPasteOpenHidden = (node: Element | TreeNodeProps): boolean => {
+    return !isTreeActionAllowed(TreePermission.Paste) ||
+      (storedNode === undefined || nodeTask !== 'copy') ||
+      !checkElementPermission(node.permissions, 'create')
+  }
+
+  const isPasteOnlyContentsHidden = (node: Element | TreeNodeProps): boolean => {
+    console.log(node)
+    return isPasteOpenHidden(node) ||
+      node.type === 'folder' ||
+      node.isLocked
+      // storedNode?.type !== node.type //TODO: i guess we need to compare the className aswell
```

Comment:
Yes for paste only contents we need to compare the class name

Reply:
Funfact: in the old UI we don`t check the className ... you can select `paste only content` from a `car` to a `event`

Reply:
Paste only contents here does not work